datafusion/physical-plan/src/memory.rs

```diff
@@ -207,6 +208,20 @@ impl MemoryExec {
     /// [`EquivalenceProperties`], we can keep track of these equivalences
     /// and treat `a ASC` and `b DESC` as the same ordering requirement.
     pub fn with_sort_information(mut self, sort_information: Vec<LexOrdering>) -> Self {
+        // All sort expressions must refer to the projected schema
+        debug_assert!({
```

Comment:
Rather than a debug assert, I think it is worth considering  changing `with_sort_information` to `try_with_sort_information` and returning a `Result<Self>` with a real error such as which columns wasn't present, )

Reply:
Done. I have also insert the logic into `try_with_sort_information()` since I've observed that ExecutionPlan API's taking `PhysicalExpr`'s as parameter refer to the input schema when that operator has a built-in projection like this `MemoryExec`. I believe we should follow this convention.

Reply:
makes sense -- thank you